tasks/licenses.py

```diff
@@ -0,0 +1,80 @@
+import json
+import pathlib
+import time
+from contextlib import closing
+from io import StringIO
+
+import httpx
+
+LATEST_API = "https://api.github.com/repos/spdx/license-list-data/releases/latest"
+LICENSES_URL = (
+    "https://raw.githubusercontent.com/spdx/license-list-data/v{}/json/licenses.json"
+)
+EXCEPTIONS_URL = (
+    "https://raw.githubusercontent.com/spdx/license-list-data/v{}/json/exceptions.json"
+)
+
+
+def download_data(url):
+    for _ in range(600):
+        try:
+            response = httpx.get(url)
+            response.raise_for_status()
+        except Exception:
+            time.sleep(1)
+            continue
+        else:
+            return json.loads(response.content.decode("utf-8"))
+
+    message = "Download failed"
+    raise ConnectionError(message)
+
+
+def main():
+    latest_version = download_data(LATEST_API)["tag_name"][1:]
+
+    licenses = {}
+    for license_data in download_data(LICENSES_URL.format(latest_version))["licenses"]:
+        license_id = license_data["licenseId"]
+        deprecated = license_data["isDeprecatedLicenseId"]
+        licenses[license_id.lower()] = {"id": license_id, "deprecated": deprecated}
+
+    exceptions = {}
+    for exception_data in download_data(EXCEPTIONS_URL.format(latest_version))[
+        "exceptions"
+    ]:
+        exception_id = exception_data["licenseExceptionId"]
+        deprecated = exception_data["isDeprecatedLicenseId"]
+        exceptions[exception_id.lower()] = {
+            "id": exception_id,
+            "deprecated": deprecated,
+        }
```

Comment:
2214401a1f2f93ddded4f636f30aa04cd2209601 adds tests around key casing, since the value is now typed as a TypedDict, mypy will detect if new keys appear or others disappear